docs/scalar_tensor.rst

```diff
@@ -72,7 +72,95 @@ Declaration
         # ERROR: cannot create tensor after any tensor accesses from the Python-scope!
 
 
-Attribute
+Accessing components
+--------------------
+
+.. attribute:: a[p, q, ...]
```

Comment:
nit: When written like this, it gives the reader an impression that a tensor must have at least two dimensions. I feel like this is such a common operation that we don't need to document it very formally. Maybe we should just show some examples? FYI here's `numpy`'s array indexing doc: https://numpy.org/devdocs/reference/arrays.indexing.html#indexing. It's too fancy for us, but it only shows examples...

Reply:
What should we do then? To write `a[obj]` and say `obj` is a `tuple`?

Reply:
Maybe just say "You can access an element of the Taichi tensor by an index", then shows your examples?

Reply:
Ack for adding a brief desc. But note that this is put in `API Reference` section, if user want a more tutorial version, they will go for `tensor_matrix.rst` in `Overview` section.

Reply:
> To write a[obj] and say obj is a tuple

Then let's do this, since it's more generic.